graph-gateway/src/network/indexers_cost_model_resolver.rs

```diff
@@ -0,0 +1,60 @@
+//! Resolves the cost models for the indexers' deployments.
+//!
+//! The cost models are fetched from the indexer's cost URL.
+
+use std::collections::HashMap;
+
+use thegraph_core::types::DeploymentId;
+use url::Url;
+
+use crate::{indexers, indexers::cost_models::CostModelSource};
+
+/// Error type for cost model resolution.
+#[derive(Debug, Clone, thiserror::Error)]
+enum ResolutionError {
+    /// The cost model couldn't be fetched.
+    #[error("The cost model couldn't be fetched: {0}")]
+    FetchFailed(String),
+}
+
+/// Resolve the indexers' cost models sources and compile them into cost models.
+pub struct CostModelResolver {
+    client: reqwest::Client,
+}
+
+impl CostModelResolver {
+    /// Creates a new [`CostModelResolver`] with the given HTTP client.
+    pub fn new(client: reqwest::Client) -> Self {
+        Self { client }
+    }
+
+    /// Fetches the cost model sources for the given deployments from the indexer.
+    ///
+    /// Returns a map of deployment IDs to the retrieved cost model sources. If certain deployment
+    /// ID's cost model fetch fails, the corresponding value in the map is `None`.
+    pub async fn resolve(
+        &self,
+        indexer_cost_url: Url,
+        indexer_deployments: &[DeploymentId],
+    ) -> HashMap<DeploymentId, CostModelSource> {
+        let sources =
+            match indexers::cost_models::query(&self.client, indexer_cost_url, indexer_deployments)
+                .await
+                .map_err(|err| ResolutionError::FetchFailed(err.to_string()))
+            {
+                Ok(sources) => sources,
+                Err(err) => {
+                    tracing::debug!("Failed to fetch cost models: {err}");
```

Comment:
feels a bit redundant to have a message like `Failed to fetch cost models: The cost model couldn't be fetched: ...`

Reply:
I will review the error messages 👍🏼

Reply:
Resolved in [`bacfaa5` (#706)](https://github.com/edgeandnode/gateway/pull/706/commits/bacfaa57051c99ee94da3e291019f91bbd10d74c)